MatrixSDK/MXRestClient.m

```diff
@@ -222,6 +222,54 @@ - (NSString*)loginFallback;
 }
 
 
+#pragma mark - password update operation
+
+/**
+ Update the account password.
+ 
+ @param oldPassword the current password to update.
+ @param newPassword the new password.
+ @param success A block object called when the operation succeeds.
+ @param failure A block object called when the operation fails.
+ 
+ @return a MXHTTPOperation instance.
+ */
+- (MXHTTPOperation*)changePassword:(NSString*)oldPassword with:(NSString*)newPassword
+                           success:(void (^)())success
+                           failure:(void (^)(NSError *error))failure
+{
+    // sanity check
+    if (!oldPassword || !newPassword)
+    {
+        NSError* error = [NSError errorWithDomain:@"Invalid params" code:500 userInfo:nil];
+        
+        failure(error);
+        return nil;
+    }
+    
+    NSDictionary *authparameters = @{
+                                     @"type": kMXLoginFlowTypePassword,
+                                     @"user": self.credentials.userId,
+                                     @"password": oldPassword,
+                                     };
+    
+    NSDictionary *parameters = @{
+                                 @"auth": authparameters,
```

Comment:
FTR, authparameters  is not required here. You can use literals inside literals - ie, @{} inside a @{} and so on.